### PR TITLE
Vet clinic database: query multiple tables

### PR DESCRIPTION
**Pull request by vanina123**
**In this pull request the objectives are**

- [ ] Use primary key & foreign key mechanism for joining tables.
- [ ] Query multiple tables.
- [ ] Prepare complex queries that answer analytical questions.

**below are the queries and their screenshots**

- What animals belong to Melody Pond?

owner

- List of all animals that are pokemon (their type is Pokemon).

spc

- List all owners and their animals, remember to include those that don't own any animal.

mal

- How many animals are there per species?

cnt

- List all Digimon owned by Jennifer Orwell.

full

- List all animals owned by Dean Winchester that haven't tried to escape.

hero

- Who owns the most animals

mel

### CODE REVIEW

File: data.sql [hunk 1]
```diff
@@ -12,3 +12,48 @@ INSERT INTO animals VALUES (5, 'Charmander', '2020-02-08', 0, false, -11),
...
+
+
+-- day 3
+-- Insert the following data into the owners table:
+-- Sam Smith 34 years old.
+-- Jennifer Orwell 19 years old.
+-- Bob 45 years old.
+-- Melody Pond 77 years old.
+-- Dean Winchester 14 years old.
+-- Jodie Whittaker 38 years old.
```
Comment (Shedrack-Sunday) [inline data.sql]: -  [ ] I suggest you take this portion of the comments from your code. Details like this should be in your README and pull request. This makes sure your codebase is not clustered and looks very simple, clean yet professional 👍

File: data.sql [hunk 2]
```diff
@@ -12,3 +12,48 @@ INSERT INTO animals VALUES (5, 'Charmander', '2020-02-08', 0, false, -11),
...
+-- Pokemon
+-- Digimon
+INSERT INTO species (name)
+VALUES ('Pokemon'),
+('Digomon');
+
+
+-- Modify your inserted animals so it includes the species_id value:
+-- If the name ends in "mon" it will be Digimon
+-- All other animals are Pokemon
```
Comment (Shedrack-Sunday) [inline data.sql]: - [ ] Kindly effect changes as in my comments on line`24` above 👍

File: data.sql [hunk 3]
```diff
@@ -12,3 +12,48 @@ INSERT INTO animals VALUES (5, 'Charmander', '2020-02-08', 0, false, -11),
...
+
+UPDATE animals SET species_id = 2 WHERE name LIKE '%mon';
+UPDATE animals SET species_id = 1 WHERE species_id IS NULL;
+
+-- Modify your inserted animals to include owner information (owner_id):
+-- Sam Smith owns Agumon.
+-- Jennifer Orwell owns Gabumon and Pikachu.
+-- Bob owns Devimon and Plantmon.
+-- Melody Pond owns Charmander, Squirtle, and Blossom.
+-- Dean Winchester owns Angemon and Boarmon.
```
Comment (Shedrack-Sunday) [inline data.sql]: - [ ] Kindly effect changes as in my comments on line`24` above 👍

File: schema.sql [hunk 4]
```diff
@@ -18,4 +18,49 @@ CREATE TABLE animals (
     neutred BOOLEAN,
     weight_kg DECIMAL,
     species VARCHAR(255)
+);
+
+-- day 3
+-- Create a table named owners with the following columns:
+-- id: integer (set it as autoincremented PRIMARY KEY)
+-- full_name: string
+-- age: integer
```
Comment (Shedrack-Sunday) [inline schema.sql]: - [ ] Kindly effect changes as I commented on line`24` in the `data.sql` file👍

File: schema.sql [hunk 5]
```diff
@@ -18,4 +18,49 @@ CREATE TABLE animals (
...
+CREATE TABLE species (
+    id serial PRIMARY KEY,
+    name VARCHAR(255) not null
+);
+
+-- Modify animals table:
+-- Make sure that id is set as autoincremented PRIMARY KEY
+-- Remove column species
+-- Add column species_id which is a foreign key referencing species table
+-- Add column owner_id which is a foreign key referencing the owners table
```
Comment (Shedrack-Sunday) [inline schema.sql]: - [ ] Kindly effect changes as in my comments on line`27` above 👍

File: queries.sql [hunk 6]
```diff
@@ -19,6 +19,7 @@ SELECT * FROM animals WHERE name Like '%mon';
 --  Find all animals with a weight between 10.4kg and 17.3kg (including the animals with the weights that equals precisely 10.4kg or 17.3kg)
 SELECT * FROM animals WHERE weight_kg BETWEEN 10.4 AND 17.3 ;
 
+-- day 2 question
 -- Inside a transaction update the animals table by setting the species column to unspecified. Verify that change was made. Then roll back the change and verify that the species columns went back to the state before the transaction.
 - BEGIN TRANSACTION;
 - UPDATE animals SET species = 'unspecified';
```
Comment (Shedrack-Sunday) [inline queries.sql:25]: - [ ] Please avoid too many comments in your code. As stated in other files, details of your codebase should be in your README and pull request and not in the codebase. Kindly adhere to this to build brain muscle in writing clean code 👍

**Review by Shedrack-Sunday — CHANGES_REQUESTED** [review]
Hello @vanina123      👋 

# CHANGES REQUIRED ♻️

Great work on the project 👏
You've done extremely well so far in this project,  However, there are some adjustments you need to make to move to the next project. 💯



## Highlights   
- Good use of GitHub features ✔️  
- Nice use of SQL queries✔️ 

 
# Required Changes  🌵 

- [ ] You made some efforts in the implementations which I must say is a job well done. However, there are some issues  I suggest   you make In line with the requirements  

_Check the comments for a detailed review 

# Optional suggestions 

Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing really improves your application.

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.

As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Review by KDlamini — APPROVED** [review]
### Hi @vanina123 🙋‍♀️,  

Excellent work! 💯 
You have done a great job implementing all the project requirements for querying multiple tables. 
Your feature is complete! There is nothing else to say other than... it's time to merge it. :shipit: 
Congratulations!!!🎊 🥳 

## Highlights 🌞

✔️  Good PR title and description
✔️  Well documented README 
✔️  All SQL commands are in their respective files
✔️  Code is very clean and organized

# Status: Approved ✔️ 

LeonardoInGatsby

Cheers 🥂  and Happy coding!!! 👯 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me @KDlamini  in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**